packages/expo/cli/start/doctor/ngrok/ExternalModule.ts

```diff
@@ -0,0 +1,194 @@
+    private promptMessage: (pkgName: string) => string
+  ) {}
+
+  /** Resolve the globally or locally installed instance, or prompt to install. */
+  async resolveAsync({
+    prefersGlobalInstall,
+    ...options
+  }: ResolvePromptOptions = {}): Promise<IModule> {
+    try {
+      return (
+        this.getVersioned() ??
+        this.installAsync({
+          ...options,
+          shouldGloballyInstall: prefersGlobalInstall,
+        })
+      );
+    } catch (error: any) {
+      if (error instanceof ExternalModuleVersionError) {
+        // If the module version in not compliant with the version range,
+        // we should prompt the user to install the package where it already exists.
+        return this.installAsync({
+          ...options,
+          shouldGloballyInstall: error.shouldGloballyInstall ?? prefersGlobalInstall,
+        });
+      }
+      throw error;
+    }
+  }
+
+  /** Prompt the user to install the package and try again. */
+  async installAsync({
+    shouldPrompt = true,
+    autoInstall,
+    shouldGloballyInstall,
+  }: InstallPromptOptions = {}): Promise<IModule> {
+    const packageName = [this.pkg.name, this.pkg.versionRange].join('@');
+    if (!autoInstall) {
+      // Delay the prompt so it doesn't conflict with other dev tool logs
+      await delayAsync(100);
+    }
+    const answer =
+      autoInstall ||
+      (shouldPrompt &&
+        (await confirmAsync({
+          message: this.promptMessage(packageName),
+          initial: true,
+        })));
+    if (answer) {
+      Log.log(`Installing ${packageName}...`);
+
+      // Always use npm for global installs
+      const packageManager = shouldGloballyInstall
+        ? new PackageManager.NpmPackageManager({
+            cwd: this.projectRoot,
+            log: Log.log,
+            silent: !EXPO_DEBUG,
+          })
+        : PackageManager.createForProject(this.projectRoot, {
+            silent: !EXPO_DEBUG,
+          });
+
+      try {
+        if (shouldGloballyInstall) {
+          await packageManager.addGlobalAsync(packageName);
+        } else {
+          await packageManager.addDevAsync(packageName);
+        }
+        Log.log(`Installed ${packageName}`);
+      } catch (e) {
+        e.message = `Failed to install ${packageName} ${
+          shouldGloballyInstall ? 'globally' : 'locally'
+        }: ${e.message}`;
+        throw e;
+      }
+      return await this.resolveAsync({ shouldPrompt: false });
+    }
+
+    throw new CommandError(
+      'EXTERNAL_MODULE_AVAILABILITY',
+      `Please install ${packageName} and try again`
+    );
+  }
+
+  /** Get the module. */
+  get(): IModule | null {
+    try {
+      return this.getVersioned();
+    } catch {
+      return null;
+    }
+  }
+
+  /** Get the module, throws if the module is not versioned correctly. */
+  getVersioned(): IModule | null {
+    if (!this.instance) {
+      this.instance = this._resolveModule(true) ?? this._resolveModule(false);
+    }
```

Comment:
```suggestion
    this.instance ??= this._resolveModule(true) ?? this._resolveModule(false);
```

Reply:
Logical nullish assignment operator is Node 15+, I think we're still trying to support Node 14 (and 12 for a couple of months)?

Reply:
It doesn't matter what the target Node version is, this is TypeScript

Reply:
Oh right, got it confused with other places where we had to dial back on the future features because JS.